### PR TITLE
Fix using `.A` instead of `.toarray()`

### PR DESCRIPTION
**Pull request by michalk8**

### CODE REVIEW

File: tests/test_gpcca.py [hunk 1]
```diff
@@ -524,6 +525,7 @@ def test_gpcca_brandts_sparse_is_not_densified(self, P: np.ndarray, sd: np.ndarr
         with pytest.raises(ValueError, match=r"Sparse implementation is only available for `method='krylov'`."):
             GPCCA(csr_matrix(P), eta=sd, method="brandts").optimize(3)
 
+    @pytest.mark.skipif(Version(np.__version__) >= Version("2"), reason="Eigenvalue mismatch.")
```
Comment (msmdev) [inline tests/test_gpcca.py:528]: So you skip this test if numpy version is >=2? Why is it ok for newer numpy versions to fail this test?

File: tests/conftest.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def bdc(q: np.ndarray, p: np.ndarray, sparse: bool = True) -> Tuple[np.ndarray,
     mu[0] = 1.0
     mu[1:] = np.cumprod(p[:-1] / q[1:])
 
-    return (P if sparse else P.A), mu / np.sum(mu)
+    return (P if sparse else P.toarray()), mu / np.sum(mu)
```
Comment (msmdev) [inline tests/conftest.py:73]: What was the problem here, exactly?

**Review by msmdev — COMMENTED** [review]
I'm not really getting the motivation/justification for your changes here. Could you please explain the reasoning behind them to me? Thank you!

File: tests/conftest.py (hunk 2 above)
Comment (michalk8) [reply tests/conftest.py:73]: This is because `scipy` deprecated `.A` and `.toarray()` should be used in favor.

File: tests/test_gpcca.py (hunk 1 above)
Comment (michalk8) [reply tests/test_gpcca.py:528]: I've been trying to figure out why new `numpy` version causes this test to fail, but was unable to figure it out. I will try later do a bisect on the commits, see exactly why, but was thinking in the meantime we could just skip it.

**Comment by michalk8** [comment]
> I'm not really getting the motivation/justification for your changes here. Could you please explain the reasoning behind them to me? Thank you!

Regarding the changes in the docs, sphinx was throwing some warning, so I corrected the template and `conf.py` - no warnings now!

File: tests/test_gpcca.py (hunk 1 above)
Comment (michalk8) [reply tests/test_gpcca.py]: I've reverted this, will leave this for a future PR to investigate.

**Comment by msmdev** [comment]
There are currently a lot of strange issues:

- failing sphinx during lint: "TypeError: can only concatenate str (not "_StrPath") to str")
- strange errors for py3.10-noslepc and py3.9-slepc: "ERROR: InvocationError for command /home/runner/work/pyGPCCA/pyGPCCA/.tox/py3.10-noslepc/bin/python -m pytest --cov --cov-append --cov-report=term-missing --cov-config=/home/runner/work/pyGPCCA/pyGPCCA/tox.ini --ignore docs/ -vv (exited with code 1)"
- the tests py3.11-noslepc and py3.12-noslepc still fail because of the known issues with test_sort_real_schur() and new numpy versions
- the macos-latest-test pretends to succeed but actually fails with "ERROR: Cannot locate the conda executable."

Honestly, I don't understand what is going on here...

@michalk8 maybe you have an idea?

**Comment by michalk8** [comment]
Fixed:
- failing sphinx during lint: "TypeError: can only concatenate str (not "_StrPath") to str")
- strange errors for py3.10-noslepc and py3.9-slepc: "ERROR: InvocationError for command /home/runner/work/pyGPCCA/pyGPCCA/.tox/py3.10-noslepc/bin/python -m pytest --cov --cov-append --cov-report=term-missing --cov-config=/home/runner/work/pyGPCCA/pyGPCCA/tox.ini --ignore docs/ -vv (exited with code 1)"
- the macos-latest-test pretends to succeed but actually fails with "ERROR: Cannot locate the conda executable."

As for the failing tests with `numpy>=2.0`, I unfortunately have no idea; will try to go over the changelog and see what might've broken the tests.

In the future, I plan refactor/modernize the installation and building of docs.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -30,29 +30,36 @@ jobs:
...
-            uses: codecov/codecov-action@v3
+            uses: codecov/codecov-action@v4
             with:
+                fail_ci_if_error: false
                 files: ./coverage.xml
-                flags: unittests
                 name: ${{ matrix.os }}-${{ matrix.python }}-${{ matrix.slepc }}
-                env_vars: OS,PYTHON
-                fail_ci_if_error: false
+                token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (michalk8) [inline .github/workflows/ci.yml:64]: @msmdev could you please create and add this token to the repo secrets?

**Comment by msmdev** [comment]
This difference is quite interesting. Maybe it is a result of differences in system architecture? Do you know, on what type of processors the MacOS tests are running: Intel (as Linux) or ARM?

File: .github/workflows/ci.yml (hunk 3 above)
Comment (msmdev) [reply .github/workflows/ci.yml:64]: Seems that it was already there in the repo secrets...

**Pull request merged**